### PR TITLE
Adding unit tests for CognitoSigninManager and CognitoUserManager

### PR DESCRIPTION
**Pull request by assyadh**
*Issue #, if available:* DOTNET-3181

*Description of changes:* 
Adding 44 unit tests overall for the SigninManager and UserManager
Switched a few of the new managers/stores methods to virtual for test-ability. 

+ Fixed one bug on user sessions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/Amazon.AspNetCore.Identity.Cognito/CognitoSigninManager.cs [hunk 1]
```diff
@@ -256,7 +256,7 @@ public async Task<SignInResult> RespondToTwoFactorChallengeAsync(string code, bo
             // Responding to the Cognito challenge.
             await _userManager.RespondToTwoFactorChallengeAsync(user, code, twoFactorInfo.CognitoAuthenticationWorkflowId).ConfigureAwait(false);
 
-            if (user.SessionTokens == null && !user.SessionTokens.IsValid())
+            if (user.SessionTokens == null || !user.SessionTokens.IsValid())
```
Comment (assyadh) [inline src/Amazon.AspNetCore.Identity.Cognito/CognitoSigninManager.cs]: Fixing a bug, The user either has valid tokens or nothing.

File: test/Amazon.AspNetCore.Identity.Cognito.Test/Amazon.AspNetCore.Identity.Cognito.Test.csproj [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
```
Comment (assyadh) [inline test/Amazon.AspNetCore.Identity.Cognito.Test/Amazon.AspNetCore.Identity.Cognito.Test.csproj]: Targeting 2.1 as it is LTS.

File: test/Amazon.AspNetCore.Identity.Cognito.Test/CognitoUserManagerTest.cs [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        {
+            userStoreMock.Setup(mock => mock.RemoveClaimsAsync(It.IsAny<CognitoUser>(), It.IsAny<IEnumerable<Claim>>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(IdentityResult.Success)).Verifiable();
+            var output = await userManager.RemoveClaimsAsync(GetCognitoUser(), new List<Claim>()).ConfigureAwait(false);
+            Assert.Equal(IdentityResult.Success, output);
+            userStoreMock.Verify();
+        }
+
+        #region ExceptionTests
+
+        [Fact]
```
Comment (sstevenkang) [inline test/Amazon.AspNetCore.Identity.Cognito.Test/CognitoUserManagerTest.cs]: is it really worth making these individual tests?  Why not make it into a single or a few grouped tests?

Comment (assyadh) [reply test/Amazon.AspNetCore.Identity.Cognito.Test/CognitoUserManagerTest.cs]: I think it's cleaner to split the tests, this way the mocking behavior is more readable.

**Pull request merged**